cxx/pclean/schema_helper.cc

```diff
@@ -25,14 +26,36 @@ void PCleanSchemaHelper::compute_domains_for(const std::string& name) {
   std::vector<std::string> annotated_ds;
   PCleanClass c = schema.classes[name];
 
-  for (const auto& v: c.vars) {
+  // Recursively maps the indices of class "name" (and ancestors) in relation
```

Comment:
Can all of this new code go into a new method, compute_ref_indices() ?  It would be great if compute_domains_cache() could stick to its original mission.

Reply:
Done (not sure which I prefer here, combining them saved some duplicate code and ensured that the domains cache was populated first, which the caller now has to ensure).